signac/contrib/migration/__init__.py

```diff
@@ -5,102 +5,135 @@
 
 import os
 import sys
-from contextlib import contextmanager
+import warnings
 
 from filelock import FileLock
 from packaging import version
 
-from ...common.config import get_config
+from ...common.config import get_config, load_config
 from ...version import SCHEMA_VERSION, __version__
-from .v0_to_v1 import migrate_v0_to_v1
+
+# To be uncommented when switching to version 2.
+# from .v0_to_v1 import _load_config_v1, _migrate_v0_to_v1
+from .v0_to_v1 import _migrate_v0_to_v1
 
 FN_MIGRATION_LOCKFILE = ".SIGNAC_PROJECT_MIGRATION_LOCK"
 
 
-MIGRATIONS = {
-    ("0", "1"): migrate_v0_to_v1,
+# Config loaders must be functions whose first argument is a directory from
+# which to read configuration information. If the logic for loading a schema
+# has not changed from previous versions, it need not be added to this dict.
+# This dictionary only needs to contains all unique loaders in signac's history
+# to ensure that any prior config may be loaded for migration.
```

Comment:
I do not think we need to support all loaders, but only those that are needed for supported migration paths.
```suggestion
# This dictionary only needs to contain all unique loaders for schema
# versions that are supported as starting points for migration.
```